stdlib/public/core/Arrays.swift.gyb

```diff
@@ -2021,31 +2021,38 @@ extension _ArrayBufferProtocol {
   internal mutating func _arrayAppendSequence<S : Sequence>(
     _ newItems: S
   ) where S.Iterator.Element == Element {
+    
+    // this function is only ever called from append(contentsOf:)
+    // which should always have exhausted its capacity before calling
+    _sanityCheck(count == capacity)
+    var newCount = self.count
 
+    // there might not be any elements to append remaining,
+    // so use an iterator+while loop to check, then increase
```

Comment:
Figured it's worth calling out here because I spent a few minutes trying to replace the original `while` loop with a `for` before conceding `while` was better in this case. There are other places in the std lib I spot sometimes where `for` could have been used but isn't so wanted to flag this ain't one of them...